File: QuantConnect.AlphaStream/Requests/GetAlphaPricesRequest.cs
using QuantConnect.AlphaStream.Infrastructure;
using QuantConnect.AlphaStream.Models;
using RestSharp;

namespace QuantConnect.AlphaStream.Requests
{
    /// <summary>
    /// Fetch Alpha historical listing prices.
    /// </summary>
    [Endpoint(Method.GET, "alpha/{id}/prices")]
    public class GetAlphaPricesRequest : AttributeRequest<Price>
    {
        /// <summary>
        /// Unique id hash of an Alpha published to the marketplace.
        /// </summary>
        [PathParameter("id")]
        public string Id { get; set; }

        /// <summary>
        /// Starting position for the search query. With very large datasets the results can be paginated and fetched in chunks starting from start.
        /// </summary>
        [QueryParameter("start")]
        public int Start { get; set; } = 0;
    }
}

File: AlphaStream/Requests/GetAlphaPricesRequest.py
class GetAlphaPricesRequest:
    """ Fetch an alpha price history in batches, starting from `start` for a maximum of 1000 values """

    def __init__(self, alphaId, start = 0):
        self.Id = alphaId
        self.Start = start
        self.Endpoint = "alpha/{}/prices".format(alphaId)
        
    def GetPayload(self):
        payload = {
            "id" : self.Id,
            "start" : self.Start
        }
        return payload